vm/src/obj/objbytearray.rs

```diff
@@ -563,6 +564,23 @@ impl PyByteArray {
     fn decode(zelf: PyRef<Self>, args: DecodeArgs, vm: &VirtualMachine) -> PyResult<PyStringRef> {
         bytes_decode(zelf.into_object(), args, vm)
     }
+
+    #[pymethod(magic)]
+    fn reduce_ex(&self, _proto: usize, vm: &VirtualMachine) -> (PyClassRef, PyTuple) {
+        self.reduce(vm)
+    }
+
+    #[pymethod(magic)]
+    fn reduce(&self, vm: &VirtualMachine) -> (PyClassRef, PyTuple) {
```

Comment:
I find that we have to impl __reduce_ex__ even we don't need to check the protocol, if I remove this function than __reduce__ will not be called also an error happens when pickling. Should we get it sorted? Any idea?